src/annotator/test/config-test.js

```diff
@@ -1,86 +1,213 @@
 'use strict';
 
-var config = require('../config');
-var unroll = require('../../shared/test/util').unroll;
-
-describe('annotator configuration', function () {
-  var fakeScriptConfig;
-
-  function fakeQuerySelector(selector) {
-    if (selector === 'link[type="application/annotator+html"]') {
-      return {href: 'app.html'};
-    } else if (selector === 'script.js-hypothesis-config' &&
-               fakeScriptConfig) {
-      return {textContent: fakeScriptConfig};
-    } else {
-      return null;
-    }
-  }
-
-  var fakeWindowBase = {
+var proxyquire = require('proxyquire');
+
+var fakeSettings = sinon.stub();
+var fakeExtractAnnotationQuery = {};
+
+var config = proxyquire('../config', {
+  '../shared/settings': fakeSettings,
+  './util/extract-annotation-query': fakeExtractAnnotationQuery,
+});
+var sandbox = sinon.sandbox.create();
+
+function fakeWindow() {
+  return {
     document: {
-      querySelector: fakeQuerySelector,
-      querySelectorAll: function (selector) {
-        var match = fakeQuerySelector(selector);
-        return match ? [match] : [];
-      },
+      querySelector: sinon.stub().returns({href: 'LINK_HREF'}),
     },
-    location: {hash: ''},
+    location: {href: 'LOCATION_HREF'},
   };
+}
+
+describe('src.annotator.config', function() {
+  beforeEach('stub console.warn()', function() {
+    sandbox.stub(console, 'warn');
+  });
+
+  beforeEach('reset fakeSettings', function() {
+    fakeSettings.reset();
+    fakeSettings.returns({});
+  });
 
-  beforeEach(function () {
-    fakeScriptConfig = '';
+  beforeEach('reset fakeExtractAnnotationQuery', function() {
+    fakeExtractAnnotationQuery.extractAnnotationQuery = sinon.stub();
   });
 
-  it('reads the app src from the link tag', function () {
+  afterEach('reset the sandbox', function() {
+    sandbox.restore();
+
+  });
+
+  it("returns the application/annotator+html <link>'s href as options.app", function() {
     var linkEl = document.createElement('link');
     linkEl.type = 'application/annotator+html';
-    linkEl.href = 'https://test.hypothes.is/app.html';
+    linkEl.href = 'http://example.com/link';
     document.head.appendChild(linkEl);
-    assert.deepEqual(config(window), {
-      app: linkEl.href,
-    });
+
+    assert.equal(config(window).app, linkEl.href);
```

Comment:
Here I'm using the real `window` as I wanted to test this one behaviour in integration with the real `window.document`. The rest of the tests don't need to test that this part (finding the `<link>` element in the document) really works and it's more convenient and probably faster for them to use a fake window.